### DISCUSSION TITLE
How do I log in as root?

### DISCUSSION THREAD

**Discussion by lucmp** [SpiralLinux/SpiralLinux-project#269] [Q&A] [answered]
Hi.
I am testing Spiral Linux, there is a good chance I will like it.
I am used to logging in as root in Debian, also did it in Slackware and Puppy, for 17 years all combined. But I can't find a way to do that in Spiral. Can you please tell me how to unlock it?
I don't mean the terminal. I mean the full graphical DE session.

**Comment by geckolinux** [comment]
Hi there, first you will need to set a root password, and then the instructions to login graphically would depend on which edition you're using.

**Reply by lucmp** [reply]
I have the Builder edition. I immediately installed LXDE (Openbox) and may or may not keep IceWM.

**Reply by geckolinux** [reply, marked as answer]
In that case after setting the root password you can login as root with the LightDM login manager by selecting "Other..." from the user list.

> and may or may not keep IceWM

Yep, the idea with the Builder edition is to just offer IceWM as the bare minimum required to launch the core GUI configuration applications like Synaptic, and from there install a full-featured desktop environment. I definitely wouldn't recommend IceWM for normal usage.

**Reply by lucmp** [reply]
Yeah, well, I think IceWM is better than all the other DEs you are shipping with Spiral. But it's all a matter of taste, right? Thank God it's Linux and we can choose. :-) 

Thank you very much for the answer! I honestly thought you wouldn't tell me. Most people are very adamant about not surrendering that information.

**Reply by geckolinux** [reply]
No problem, you obviously know what's best for your needs and preferences. I just try to discourage inexperienced users from using the SpiralLinux Builder edition as-is as if it were a dedicated lightweight edition; it's just intended as a starting point for customization.

**Reply by lucmp** [reply]
Erm, sorry, me again.
I did what you told me and now I can log in as root.
But the login screen never shows root. I have to click the regular user name, open the drop-down, select "Other" then type root. 
Long procedure, kind of annoying. Do you know how to add root and make it default?

**Reply by geckolinux** [reply]
I'm not sure if LightDM will allow that (they do discourage running as root). But you can try editing `/etc/lightdm/users.conf` and setting `minimum-uid=0`

**Reply by lucmp** [reply]
That didn't work.
My old Debian has minimum-uid=500 too, but it doesn't have the drop-down, just an empty entry field.
Maybe they changed it? :-(

**Reply by geckolinux** [reply]
But did you try changing it from `500` to `minimum-uid=0` ?

**Reply by lucmp** [reply]
Yes, I did that too.
I also tried SDDM and it seems to be worse so I restored from a snapshot.
But I still can't get what I want with lightdm.

**Reply by lucmp** [reply]
I seem to have found a way, but I can't save the changes because the system is read-only and I cannot rollback.
Please see attachment.
Do you have any ideas?

![rollback](https://github.com/SpiralLinux/SpiralLinux-project/assets/2882969/82424745-fb0f-4186-b223-fcac218b32ed)

**Reply by geckolinux** [reply]
https://github.com/SpiralLinux/SpiralLinux-project/wiki#working-with-btrfs-snapshots-and-rollbacks
